repository/src/test/java/org/alfresco/repo/lock/JobLockServiceTest.java

```diff
@@ -549,19 +548,19 @@ public void runGetLockWithCallback(int t)
             
             if (callback == null) throw new IllegalStateException();
 
-            waitForAssertion(of(100, MILLIS), () -> {
+            TestHelper.waitForMethodToFinish(of(100, MILLIS), () -> {
                 assertEquals(false,callback.released);
                 assertEquals(0,callback.getIsActiveCount());
             });
 
-            waitForAssertion(of(1, SECONDS), () -> {
+            TestHelper.waitForMethodToFinish(of(1, SECONDS), () -> {
                 assertEquals(false, callback.released);
                 assertEquals(1, callback.getIsActiveCount());
             });
 
             callback.isActive = false;
 
-            waitForAssertion(of(2, SECONDS), () -> {
+            TestHelper.waitForMethodToFinish(of(2, SECONDS), () -> {

```

Comment:
New method is not exactly the same as `waitForAssertion`. `waitForMethodToFinish` retries on *any* `Exception` while `waitForAssertion` only on the `AssertionFailedError`